pkg/driver/params.go

```diff
@@ -88,3 +98,41 @@ func NewVolumeParams(m map[string]string) (*VolumeParams, error) {
 
 	return params, nil
 }
+
+// NewSnapshotParams parses the input params and instantiates new SnapshotParams.
+func NewSnapshotParams(m map[string]string) (*SnapshotParams, error) {
+	var err error
+	params := &SnapshotParams{ // set up defaults, if any.
+		SnapSize:    100,
+		AbsSnapSize: false,
+	}
+	// parameter keys may be mistyped from the CRD specification when declaring
+	// the storageclass, which kubectl validation will not catch. Because
+	// parameter keys (not values!) are all lowercase, keys may safely be forced
+	// to the lower case.
+	m = helpers.GetCaseInsensitiveMap(&m)
+
+	size, ok := m["snapsize"]
+	if ok {
+		if strings.HasSuffix(size, "%") {
+			snapSize := size[:len(size)-1]
+			params.SnapSize, err = strconv.ParseFloat(snapSize, 64)
+			if err != nil {
+				return nil, err
+			}
+			if params.SnapSize < 1 || params.SnapSize > 100 {
+				return nil, fmt.Errorf("snapSize percentage should be between 1 and 100, found %s", size)
+			}
+		} else {
+			qty, err := resource.ParseQuantity(size)
+			if err != nil {
+				return nil, err
+			}
+			snapSize, _ := qty.AsInt64()
+			params.AbsSnapSize = true
+			params.SnapSize = float64(snapSize)
```

Comment:
should we validate it ? it should not be <= 0? sorry I missed this in my last review.

Reply:
I thought ParseQuantity can handle negative values but it turned out it cannot :sweat_smile: 
https://play.golang.org/p/Ti1Brz2MIBi